### PR TITLE
[mlir][TosaToLinalg] Lower tosa reduce operations to linalg reduce op

### PR DESCRIPTION
**Pull request by amirBish**
-lowering tosa reduce operations to linalg.reduce
op which indicates explicit reduce operation.

-the linalg reduce op doesn't have iterator
types or indexing maps instead it has a reduction
dimension which would provide the necessary
information.

### CODE REVIEW

**Comment by amirBish** [comment]
@rsuderman @NatashaKnk @eric-k256 @ftynse 
Hey, would be glad to get your feedback/review regarding this PR.
Somehow, I don't have the ability to assign you as reviewers (maybe because I don't have write access for LLVM project)

**Comment by ftynse** [comment]
> @ftynse
> Hey, would be glad to get your feedback/review regarding this PR.

Looks okay to me, but please get approval from somebody who knows TOSA.

**Review by eric-k256 — APPROVED** [review]
It looks correct to me TOSA-wise and avoids the extra work to create the maps for linalg.generic.

**Pull request merged**
